### DISCUSSION TITLE
Installing IPOPT on M1 mac Issue with HSL

### DISCUSSION THREAD

**Discussion by AnthonyDario** [coin-or/Ipopt#554] [Q&A] [answered]
Hello, I'm trying to install IPOPT on an M1 mac and I'm running into problems running the `.configure` step from the instructions. It looks like the HSL library is being built for x86 for some reason. The output of the configure script ends with:
```
...
checking for function ma27ad_ in -L/usr/local/lib -lcoinhsl   ... no
checking for function ma27ad in -L/usr/local/lib -lcoinhsl   ... no
checking for function MA27AD_ in -L/usr/local/lib -lcoinhsl   ... no
checking for function MA27AD in -L/usr/local/lib -lcoinhsl   ... no
checking for function ma27ad__ in -L/usr/local/lib -lcoinhsl   ... no
checking for function ma27ad_ in -L/usr/local/lib -lcoinhsl   ... no
checking for function MA27AD__ in -L/usr/local/lib -lcoinhsl   ... no
checking for function MA27AD_ in -L/usr/local/lib -lcoinhsl   ... no
configure: error: Provided package HSL is not working or does not contain MA27.
```

The relevant piece from `config.log` seems to be:
```
...
configure:24545: checking for function ma27ad_ in -L/usr/local/lib -lcoinhsl                  
configure:24570: gcc -o conftest -O2 -DNDEBUG    conftest.c -L/usr/local/lib -lcoinhsl    >&5
ld: warning: ignoring file /usr/local/lib/libcoinhsl.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
Undefined symbols for architecture arm64:
  "_ma27ad_", referenced from:
      _main in conftest-2123bd.old: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
...
```

Running the steps in the IPOPT installation instructions for HSL I don't see any error output. I have tried running both configure scripts with the `CC=gcc-11 CXX=g++-11` flags without success. Is there something I'm missing? Help would be much appreciated!

**Comment by svigerske** [comment]
You'll have to look into the build of the HSL library. Maybe the Fortran compiler defaults to build for macOS-x86_64 and some extra flags need to be given (`ADD_FFLAGS`, `ADD_FCFLAGS`)?
(I have no M1 mac to try this out.)

**Comment by AnthonyDario** [comment, marked as answer]
Thanks for the prompt reply! I was able to get it working with coinbrew. The command I used was `arch -x86_64 ./coinbrew build Ipopt --prefix /usr/local --test --no-prompt --verbosity=3 CC=gcc-11 CXX=g++-11`.

I did have a problem with my metis installation from homebrew though. I needed to install it using an x86 installation of homebrew. The instructions for getting this version of homebrew I found [on stackoverflow](https://stackoverflow.com/questions/64882584/how-to-run-the-homebrew-installer-under-rosetta-2-on-m1-macbook).
